packages/expo-cli/src/credentials/views/IosProvisioningProfile.ts

```diff
@@ -73,114 +68,73 @@ export class RemoveProvisioningProfile implements IView {
     if (shouldRevoke) {
       await ctx.ensureAppleCtx();
       const ppManager = new ProvisioningProfileManager(ctx.appleCtx);
-      await ppManager.revoke(selected.bundleIdentifier);
+      await ppManager.revoke(app.bundleIdentifier);
     }
   }
 }
 
 export class CreateProvisioningProfile implements IView {
-  _experienceName: string;
-  _bundleIdentifier: string;
-  _distCert: DistCert;
-  _nonInteractive: boolean;
-
-  constructor(options: ProvisioningProfileOptions) {
-    const { experienceName, bundleIdentifier, distCert } = options;
-    this._experienceName = experienceName;
-    this._bundleIdentifier = bundleIdentifier;
-    this._distCert = distCert;
-    this._nonInteractive = options.nonInteractive ?? false;
-  }
+  constructor(private app: AppLookupParams, private nonInteractive: boolean = false) {}
 
   async create(ctx: Context): Promise<ProvisioningProfile> {
     const provisioningProfile = await this.provideOrGenerate(ctx);
-    const appleTeam = ctx.hasAppleCtx()
-      ? ctx.appleCtx.team
-      : await getCredentialsFromUser(appleTeamSchema);
-    if (!appleTeam) {
-      throw new Error('Must provide a valid Apple Team Id');
-    }
-    return await ctx.ios.updateProvisioningProfile(
-      this._experienceName,
-      this._bundleIdentifier,
-      provisioningProfile,
-      appleTeam
-    );
+    return await ctx.ios.updateProvisioningProfile(this.app, provisioningProfile);
   }
 
   async open(ctx: Context): Promise<IView | null> {
     await this.create(ctx);
 
     log(chalk.green('Successfully created Provisioning Profile\n'));
-    const appCredentials = ctx.ios.credentials.appCredentials.find(
-      app =>
-        app.experienceName === this._experienceName &&
-        app.bundleIdentifier === this._bundleIdentifier
-    )!;
+    const appCredentials = await ctx.ios.getAppCredentials(this.app);
     displayIosAppCredentials(appCredentials);
     log();
     return null;
   }
 
   async provideOrGenerate(ctx: Context): Promise<ProvisioningProfile> {
-    if (!this._nonInteractive) {
+    if (!this.nonInteractive) {
       const userProvided = await askForUserProvided(provisioningProfileSchema);
       if (userProvided) {
         // userProvided profiles don't come with ProvisioningProfileId's (only accessible from Apple Portal API)
-        log(chalk.yellow('Provisioning profile: Unable to validate uploaded profile.'));
-        return userProvided;
+        log(chalk.yellow('Provisioning profile: Unable to validate specified profile.'));
+        return {
+          ...userProvided,
+          ...provisioningProfileUtils.readAppleTeam(userProvided.provisioningProfile),
```

Comment:
i noticed that we are passing in the user's Apple team info when handling the Provisioning Profile, and we weren't doing this before. Is there a new use case that requires us to get this info?

Reply:
Currently, if someone wanted to pass provisioning profile manually they needed to either login to apple developer portal or pass teamId manually https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/credentials/views/IosProvisioningProfile.ts#L101
teamId is required in www